core/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1447,6 +1429,24 @@ public QueuedMessage(ProtocolMessage msg, CompletionListener listener) {
         }
     }
 
+    /**
+     * Send all pending messages which are queue.
+     * Remove them from the queue once they are send successfully
+     * Spec: RTN19a
+     */
+    public void sendPendingQueueMessages() {
+        //RTN19a
+        if (pendingMessages.queue != null && !pendingMessages.queue.isEmpty()) {
+            for (final QueuedMessage queuedMessage : pendingMessages.queue) {
+                try {
+                    send(queuedMessage.msg, false, null);
```

Comment:
Didn't this pending message sending exist before this change?

Reply:
`send()` is used in many ways but never before to iterate through all pending messages and send them.